File: 2020_2021/tracking/get_masks.m
close all; clear all; clc;

%%
prefix = 'C:\Users\sydne\Documents\thermal_experiments\data\';
prefix  = 'G:\My Drive\Tuthill Lab Shared\Katie\thermal_experiments\data\snow_flies';

%% automate specifying the mask for the snow fly's range of motion

files = dir(prefix);
dates = files([files.isdir]);

for i=3:numel(dates)
    date = dates(i).name;
    prefix2 = fullfile(prefix, date);
    files2 = dir(prefix2);
    flies = files2([files2.isdir]);
    
    for j=3:numel(flies)     
        prefix3 = fullfile(prefix2, flies(j).name);
        files3 = dir(prefix3);
        trials = files3([files3.isdir]);
        
        for k=3:numel(trials)            
            
            prefix4 = fullfile(prefix3, trials(k).name);
            if isfile(fullfile(prefix4, 'roi.png')) || ~isfile(fullfile(prefix4, 'mask.bmp'))
                continue
            end 

            prefix4
            e1 = fullfile(prefix4, 'mask - Ellipse 1.bmp');
            e2 = fullfile(prefix4, 'mask - Ellipse 2.bmp');
            if (~isfile(e1) || ~isfile(e2))
                continue
            end
            
            ellipse1 = imread(e1);
            roi1 = imbinarize(ellipse1);
            
            ellipse2 = imread(e2);
            roi2 = imbinarize(ellipse2);
            
            mask = roi1 + roi2;
            mask(mask == 2) = 0;
            mask = mask.*roi1;
            imwrite(mask, fullfile(prefix4, 'roi.png'));
            
        end
    end
end    


%%

File: 2020_2021/tracking/all_python/get_masks.py
import os
import cv2
import pandas as pd
import numpy as np

prefix = 'G:\My Drive\Tuthill Lab Shared\Katie\thermal_experiments\2020-2021\data\snow_flies'
sessions = ['12.29.20', '12.28.20', '1.1.21', '1.2.21', '1.3.21', '1.11.21', '1.15.21', '1.16.21', '2.1.21', '2.5.21', '2.10.21', '3.1.21', '3.9.21', '3.18.21']

# for testing
prefix = '/Users/katierupp/Downloads'
sessions = ['3.1.21']

for session in sessions: 

    prefix2, snow_flies, _ = next(os.walk(os.path.join(prefix, session)))

    for fly in snow_flies:

        prefix3, trials, _ = next(os.walk(os.path.join(prefix2, fly)))

        for trial in trials:

            prefix4 = os.path.join(prefix3, trial)
            outpath = os.path.join(prefix4, 'roi.png')

            if os.path.isfile(outpath):
                continue 

            e1 = os.path.join(prefix4, 'mask - Ellipse 1.bmp')
            e2 = os.path.join(prefix4, 'mask - Ellipse 2.bmp')
            if not os.path.isfile(e1) or not os.path.isfile(e2):
                continue
            
            ellipse1 = cv2.imread(e1, 0)
            _, roi1 = cv2.threshold(ellipse1, 0, 1, cv2.THRESH_BINARY)
            
            ellipse2 = cv2.imread(e2, 0)
            _, roi2 = cv2.threshold(ellipse2, 0, 1, cv2.THRESH_BINARY)
            
            mask = roi1 + roi2
            mask[mask == 2] = 0
            mask = np.multiply(mask, roi1)
            cv2.imwrite(outpath, mask)
            